README.md

```diff
@@ -16,16 +23,178 @@ There have been, and are currently, several efforts in this area:
+- Added attribute details type [string]
+  - A type hint to indicate to the end-user what values can be provided
+- Added tag properties
+  - Lists all available properties of a web component
+- Added tag property details with name, description, default, required, type
+  - See attributes
+- Added tag methods with name, description
+  - Lists all methods or a web component
+  - Does not define parameters or type of parameters (this is for a later iteration) - for now, you can add this information to the description
+- Added tag events with name, description
+  - Lists all events the custom elements fires via `this.dispatchEvent()`
+- Added tag slots with name, description
+  - Lists all [slots](https://html.spec.whatwg.org/multipage/scripting.html#the-slot-element) in the shadow dom like `<slot name="header"></slot>`
+  - The default slot is represented via an empty name `<slot></slot>`
+- Added tag cssProperties with name, description, type
+  - Lists all [CSS custom properties](https://drafts.csswg.org/css-variables/#defining-variables) that are used within the style of a web component like `--my-css-property-name`
+  - Type are aligned with [Houdini's CSS Properties and Values API](https://drafts.css-houdini.org/css-properties-values-api)
+- Added tag cssShadowParts with name, description
+  - Lists all [CSS Shadow parts](https://www.w3.org/TR/css-shadow-parts-1/)
+
+
+## Use Cases
+
+### Editor Support
+
+Developers using custom elements should be able to get full-featured IDE support including auto-completion, hover-documentation, unknown symbol warnings, etc. These features should be available in HTML files, and in various template syntaxes via template-specific tools.
+
+### Documentation
 
 Documentation viewers should be able to display all the relevant information about a custom element, such as its tag name, attributes, properties, definition module, CSS variables and parts, etc.
 
-## Linting
+### Documentation Playground
+
+With full information about the public api of a web component tools can provide UIs to trigger/execute these apis.
+
+### Linting
+
+Linters should be able to produce warnings based on custom element definitions, such as warning if unknown elements are used in HTML templates.
+
+### Web Component Catalog
+
+A web component catalog can index packages based on the information that is available in a `custom-elements.json`.
+
+### Browser extensions
+
+A browser extension can list all used web components on a page and then query a catalog to see if meta-information is available.
+If so links to documentation, source, or ways of interacting with the elements api can be provided.
+
+## Version history
+
+### Version 1
+
+Currently implemented in VS Code. See the details for [vscode-html-languageservice](https://github.com/microsoft/vscode-html-languageservice/blob/master/docs/customData.md#custom-data-for-html-language-service).
+
+```json
```

Comment:
Not sure we need the inline demo for version one if it's going to not be considered the current version when this is published.

Reply:
Version history can be quite informative therefore I think it's worth the effort but it doesn't need to be in the main readme for sure.

Reply:
moved to a separate file 👍